packages/devextreme/js/ui/number_box/number_box.mask.js

```diff
@@ -420,7 +420,11 @@ const NumberBoxMask = NumberBoxBase.inherit({
         const isValueChanged = parsedValue !== this._parsedValue;
 
         const isDecimalPointRestricted = char === decimalSeparator && maxPrecision === 0;
-        const isUselessCharRestricted = !isTextSelected && !isValueChanged && char !== MINUS && !this._isValueIncomplete(editedText) && this._isStub(char);
```

Comment:
It's hard to believe that this condition was added without a reason. Can we research it a bit more, maybe find a with this change?

Reply:
checked. Looks what we can do this